src/plugins/navigation_embeddable/public/components/editor/navigation_embeddable_panel_editor_link.tsx

```diff
@@ -42,24 +45,85 @@ export const NavigationEmbeddablePanelEditorLink = ({
   parentDashboard?: DashboardContainer;
   dragHandleProps?: DraggableProvidedDragHandleProps;
 }) => {
+  const [dashboardError, setDashboardError] = useState<Error | undefined>();
   const parentDashboardTitle = parentDashboard?.select((state) => state.explicitInput.title);
   const parentDashboardId = parentDashboard?.select((state) => state.componentState.lastSavedId);
 
   const { value: linkLabel, loading: linkLabelLoading } = useAsync(async () => {
-    let label = link.label;
-    if (link.type === DASHBOARD_LINK_TYPE && !label) {
+    if (link.type === DASHBOARD_LINK_TYPE) {
       if (parentDashboardId === link.destination) {
-        label = parentDashboardTitle;
+        return link.label || parentDashboardTitle;
```

Comment:
nit
```suggestion
        return link.label ?? parentDashboardTitle;
```

Reply:
The `link.label` can sometimes be the empty string and is not always undefined - so, I believe `||` works better here to prevent an empty custom label. What do you think?